### DISCUSSION TITLE
Who is using uWebSockets.js?

### DISCUSSION THREAD

**Discussion by knight10-ux** [uNetworking/uWebSockets.js#765] [Q&A] [answered]
Do you have a list of companies who are using this project in production and possibly small description for what it is being used?

We are using uWebSockets.js (communication platform) for api and websockets server in prelease, but unfortunatelly cannot figure out whether it is battle tested or not. Moreover it has only one contributor, why is that if it's the fastest server for node??

It will be super helpful. Thanks in advance!

**Comment by ghost** [comment, marked as answer]
One I know for sure is bitfinex.com but I don't keep track of (or can keep track of) all companies and I'm not even legally allowed to disclose it unless they want to

**Reply by knight10-ux** [reply]
Can you share the numbers then without brand names?

**Reply by ghost** [reply]
I don't track users so there is no way to tell. I know about 10 companies who use it but it can be hundreds.

**Reply by knight10-ux** [reply]
Are there any usecases in public facing apis or just internal?

**Reply by ghost** [reply]
Well, yes. They are public. All trading on bitfinex goes via it.

**Comment by ghost** [comment]
> Moreover it has only one contributor, why is that if it's the fastest server for node??

Interesting question, for a number of different reasons. I don't know.

**Reply by knight10-ux** [reply]
I think having public endorsments from companies and at least one more contributor would increas popularity of uws by leaps and bounds. It's great that we see Bun is using it under the hood, but it's still has way to go. Anyways, I will take now the opportunity to thank you for this project.

**Reply by ghost** [reply]
But the project is not in a "growth stage", it's 7 years old and I don't really care for popularity. I would much rather see Bun explode in popularity since it is a much wider project and highlights many of the design flaws in Node.js.

**Reply by knight10-ux** [reply]
> I don't really care for popularity

So what's the purpose of the project if not wider adoption?

**Reply by ghost** [reply]
It's a niche project, it does not apply to everyone's use cases. I rather have 10 impressive commerical users than a billion amateur users

**Reply by knight10-ux** [reply]
What is an impressive commercial user in your opinion?

**Reply by ghost** [reply]
What I'm saying is I don't care about quantity I care about quality.

**Reply by knight10-ux** [reply]
imo more quantity, event on amateur level, equals more quality. Same goes for the number of great commercial success stories, since people behind those project are people too. They check how many people are using stuff and make decisions based on that. What's your take on that?

**Reply by ghost** [reply]
It becomes a catch 22 because if you intend to satisfy everyone you cannot diverge from the interfaces most assume (such as ReadableStream, IncomingMessage, EventEmitter) so to fix that you need to do what everyone else is doing but then you end up with the same crap you aimed to improve and now we are in an endless loop.

Therefore I don't target everyone I target those who need something specially tailored for their particular use case.

**Reply by ghost** [reply]
There are tons of small quirks here, which allow better performance but also brings, well, small quirks. And those quirks stand out as annoyances for most nodejs people

**Reply by knight10-ux** [reply]
This makes total sense. Thanks for info :)
